Makefile

```diff
@@ -176,135 +159,135 @@ include hack/pip-packages/Makefile
-		-t docker.io/$(DOCKER_REPOSITORY):$(REPO_REV)-devkit-amd64 \
-		--build-arg USER_ID=$(UID) \
-		--build-arg GROUP_ID=$(GID) \
-		--build-arg USER_NAME=$(USER_NAME) \
-		--build-arg GROUP_NAME=$(GROUP_NAME) \
-		--build-arg DOCKER_GID=$(DOCKER_SOCKET_GID) \
-		--build-arg BUILDARCH=amd64 \
-		--platform linux/amd64 \
-		--file $(REPO_ROOT_DIR)/Dockerfile.devkit \
-		--secret id=githubtoken,src=github-token.txt \
-		--provenance=false  \
-		--push \
-		$(REPO_ROOT_DIR)
+###### Devkit container image
+DEVKIT_IMAGE_DOCKERFILE ?= Dockerfile.devkit
+DEVKIT_IMAGE_NAME ?= mesosphere/konvoy-image-builder-devkit
+DEVKIT_IMAGE_TAG ?= $(shell cat ${DEVKIT_IMAGE_DOCKERFILE} requirements.txt requirements-devkit.txt  | sha256sum | cut -d" " -f 1)
 
-.PHONY: docker-build-amd64
-docker-build-amd64: BUILDARCH=amd64
-docker-build-amd64: devkit-amd64 konvoy-image-amd64
-	docker buildx build \
-		--file $(REPO_ROOT_DIR)/Dockerfile \
-		--build-arg BUILDARCH=amd64 \
-		--platform linux/amd64 \
-		--build-arg BASE=docker.io/$(DOCKER_REPOSITORY):$(REPO_REV)-devkit-amd64 \
-		--tag=$(DOCKER_REPOSITORY):$(REPO_REV)-amd64 \
-		--pull \
-		--provenance=false \
-		--push \
-		$(REPO_ROOT_DIR)
+.PHONY: devkit-image
+## first tries to pull an image, if doesn't exist build and push the image
+devkit-image:
+	$(call print-target)
+	docker image inspect $(DEVKIT_IMAGE_NAME):$(DEVKIT_IMAGE_TAG) &>/dev/null || \
+	docker pull $(DEVKIT_IMAGE_NAME):$(DEVKIT_IMAGE_TAG) || \
+	$(MAKE) devkit-image-amd64 devkit-image-arm64
 
-.PHONY: docker-build-arm64
-docker-build-arm64: BUILDARCH=arm64
-docker-build-arm64: devkit-arm64 konvoy-image-arm64
-	docker buildx build \
-		--file $(REPO_ROOT_DIR)/Dockerfile \
-		--build-arg BUILDARCH=arm64 \
-		--platform linux/arm64 \
-		--build-arg BASE=docker.io/$(DOCKER_REPOSITORY):$(REPO_REV)-devkit-arm64 \
-		--tag=$(DOCKER_REPOSITORY):$(REPO_REV)-arm64 \
-		--pull \
-		--push \
-		--provenance=false  \
-		$(REPO_ROOT_DIR)
+.PHONY: devkit-image-amd64
+devkit-image-amd64:
+	$(MAKE) devkit-image-build-push BUILDARCH=amd64
 
+.PHONY: devkit-image-arm64
+devkit-image-arm64:
+	$(MAKE) devkit-image-build-push BUILDARCH=arm64
 
-WHAT ?= bash
+.PHONY: devkit-image-build-push
+devkit-image-build-push: github-token.txt buildx
+	$(call print-target)
+	docker image inspect $(DEVKIT_IMAGE_NAME):$(DEVKIT_IMAGE_TAG)-$(BUILDARCH) &>/dev/null || \
+	docker pull $(DEVKIT_IMAGE_NAME):$(DEVKIT_IMAGE_TAG)-$(BUILDARCH) || \
+	docker buildx build \
+	--pull \
+	--push \
+	--build-arg BUILDARCH=$(BUILDARCH) \
+	--secret id=githubtoken,src=github-token.txt \
+	--provenance=false \
+	--platform linux/$(BUILDARCH) \
+	--file $(DEVKIT_IMAGE_DOCKERFILE) \
+	--tag=$(DEVKIT_IMAGE_NAME):$(DEVKIT_IMAGE_TAG)-$(BUILDARCH) \
+	$(dir $(DEVKIT_IMAGE_DOCKERFILE))
+
+.PHONY: devkit-image-push-manifest
+devkit-image-push-manifest: ## pushes the devkit-image
+devkit-image-push-manifest: devkit-image-amd64 devkit-image-arm64
+	$(call print-target)
+	docker manifest create \
```

Comment:
I will file separate PR in feature to add this because I want to try this after upgrading to Docker >= 23.0.x. Docker 23.0.x adds a lot of enhancements with buildx.  Adding TODO in comment to make sure we revisit it.